### PR TITLE
Adds support for ID, Class & Tag selectors to Element.querySelectorAll

### PR DESCRIPTION
**Pull request by nainar**
Addresses #111

### CODE REVIEW

**Comment by nainar** [comment]
@kristoferbaxter this is a copy of Element::querySelector, except for the return case.

**Comment by kristoferbaxter** [comment]
I'm curious if `querySelector` could be refactored to delegate functionality to `querySelectorAll` and just return the 0th value.

This would reduce the amount of code significantly. Thoughts?

File: src/test/htmlelement/querySelectorAll.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  t.deepEqual(parentDiv.querySelectorAll('DIV'), [div]);
+
+  t.deepEqual(document.querySelectorAll('div'), document.querySelectorAll('DIV'));
+  t.deepEqual(parentDiv.querySelectorAll('div'), parentDiv.querySelectorAll('DIV'));
+});
+
+test('test Element.querySelector returns the first result of Element.querySelectorAll', t => {
+  const { parentDiv } = t.context as { parentDiv: HTMLElement; div: HTMLElement };
+  let querySelectorAllResults = document.querySelectorAll('div');
+  t.true(!!querySelectorAllResults);
```
Comment (nainar) [inline src/test/htmlelement/querySelectorAll.ts]: Is there a better way for checking that this isn't null?

File: src/test/htmlelement/querySelectorAll.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  t.deepEqual(querySelectorAllResults![0], document.querySelector('div'));
```
Comment (nainar) [inline src/test/htmlelement/querySelectorAll.ts]: Is there a safer way of telling typescript that I have already asserted before that this thing isn't null?

Comment (kristoferbaxter) [reply src/test/htmlelement/querySelectorAll.ts]: You could wrap this test in a conditional, since you're certain it will pass the test will always run.

**Pull request merged**